conda/models/dist.py

```diff
@@ -17,12 +17,19 @@
 from ..common.constants import NULL
 from ..common.url import has_platform, is_url, join_url
 from ..deprecations import deprecated
+from ..history import dist_str_to_quad, strip_extension  # noqa: F401
 from .channel import Channel
 from .package_info import PackageInfo
 from .records import PackageRecord
 
 log = getLogger(__name__)
 
+deprecated.module(
+    "24.3",
+    "24.9",
```

Comment:
This is the earliest I'd be comfortable with.
`Dist` is used elsewhere outside of https://github.com/conda .
Unless it induces noticeable maintenance burden, we can also postpone the actual removal to early 2025 or so.
Marking this for `24.9` just gives us the possibility remove earlier.